src/com/microsoft/azure/documentdb/JsonSerializable.java

```diff
@@ -440,9 +440,8 @@ void onSave() {
             return c.cast(this.propertyBag);
         } else {
             // POJO
-            if (!c.isMemberClass() || !Modifier.isStatic(c.getModifiers())) {
-                throw new IllegalArgumentException(
-                        "c must be a member (not an anonymous or local) and static class.");
+            if (c.isMemberClass() && !Modifier.isStatic(c.getModifiers())) {
+                throw new IllegalArgumentException("Nested classes must be static.");
```

Comment:
Since there will be 3 places with the same check, we can put them into a private static method called "<T extends Object> void checkForValidPOJO(class<T> c)"